pkg/utils/utils.go

```diff
@@ -345,32 +345,32 @@ func GetNSFilter(obj *K8ValidNamespaces) (string, string) {
 	return key, value
 }
 
-func AddNamespaceToFilter(namespace string, obj *K8ValidNamespaces) {
-	obj.validNSList.lock.Lock()
-	defer obj.validNSList.lock.Unlock()
-	obj.validNSList.nsList[namespace] = true
+func AddNamespaceToFilter(namespace string) {
+	globalNSFilterObj.validNSList.lock.Lock()
+	defer globalNSFilterObj.validNSList.lock.Unlock()
+	globalNSFilterObj.validNSList.nsList[namespace] = true
 }
 
-func DeleteNamespaceFromFilter(namespace string, obj *K8ValidNamespaces) {
-	obj.validNSList.lock.Lock()
-	defer obj.validNSList.lock.Unlock()
-	delete(obj.validNSList.nsList, namespace)
+func DeleteNamespaceFromFilter(namespace string) {
+	globalNSFilterObj.validNSList.lock.Lock()
+	defer globalNSFilterObj.validNSList.lock.Unlock()
+	delete(globalNSFilterObj.validNSList.nsList, namespace)
 }
 
-func CheckIfNamespaceAccepted(namespace string, obj *K8ValidNamespaces, nsLabels map[string]string, nonNSK8ResFlag bool) bool {
+func CheckIfNamespaceAccepted(namespace string, nsLabels map[string]string, nonNSK8ResFlag bool) bool {
```

Comment:
It looks like you are mostly passing a `nil` for nsLabels in most of the calls to this method - in such a case should we move this as a optional parameter (third in this arg list). That way all your methods can just pass 2 args to this method except the ones that need to pass the nsLabels.